tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/RunCspellTool/RunCspellTool.cs

```diff
@@ -0,0 +1,290 @@
+        }
+
+        [McpServerTool(Name = "run-cspell"), Description("Run cspell to check for typos in the specified package path")]
+        public async Task<DefaultCommandResponse> RunCspellCheck(string packagePath)
+        {
+            try
+            {
+                logger.LogInformation("Running cspell check for package path: {packagePath}", packagePath);
+
+                if (string.IsNullOrWhiteSpace(packagePath))
+                {
+                    SetFailure(1);
+                    return new DefaultCommandResponse
+                    {
+                        ResponseError = "Package path is required"
+                    };
+                }
+
+                if (!Directory.Exists(packagePath))
+                {
+                    SetFailure(1);
+                    return new DefaultCommandResponse
+                    {
+                        ResponseError = $"Package path does not exist: {packagePath}"
+                    };
+                }
+
+                // Get the absolute path to the Invoke-Cspell.ps1 script
+                var scriptPath = GetInvokeCspellScriptPath();
+                if (string.IsNullOrEmpty(scriptPath))
+                {
+                    SetFailure(1);
+                    return new DefaultCommandResponse
+                    {
+                        ResponseError = "Could not locate Invoke-Cspell.ps1 script"
+                    };
+                }
+
+                var startTime = DateTime.UtcNow;
+                var output = await RunPowerShellScript(scriptPath, packagePath);
+                var duration = (DateTime.UtcNow - startTime).TotalMilliseconds;
+
+                // Parse the cspell output to determine success/failure and extract typos
+                var result = ParseCspellOutput(output);
+
+                return new DefaultCommandResponse
+                {
+                    Message = result.Success ? "Cspell check completed successfully" : "Cspell check found spelling errors",
+                    Result = new
+                    {
+                        Success = result.Success,
+                        TyposFound = result.TyposFound,
+                        Output = result.Output,
+                        Guidance = GetGuidanceMessage(result)
+                    },
+                    Duration = (long)duration
+                };
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error occurred while running cspell check: {packagePath}", packagePath);
+                SetFailure(1);
+                return new DefaultCommandResponse
+                {
+                    ResponseError = $"Error occurred while running cspell check: {ex.Message}"
+                };
+            }
+        }
+
+        private string GetInvokeCspellScriptPath()
+        {
+            // Try to find the script relative to the current working directory
+            var possiblePaths = new[]
```

Comment:
I think it would be better to find the git repo root via `IGitHelper.DiscoverRepoRoot` then navigate to `eng/common`. This will prevent us possibly trying to do anything at `/`.